### PR TITLE
hotplug_mem_repeat: skips memory unplug process on all rhel7 guests

### PR DESCRIPTION
**Pull request by mcasquer**
hotplug_mem_repeat: skips memory unplug process on all rhel7 guests

ID: 2054033
Signed-off-by: mcasquer <mcasquer@redhat.com>

### CODE REVIEW

**Comment by mcasquer** [comment]
@huangyum @yanan-fu please could you review this PR? Thanks!

**Comment by yanan-fu** [comment]
@mcasquer You want to disable rhel7.9 guest only or all rhel7 guest ?

**Comment by mcasquer** [comment]
> @mcasquer You want to disable rhel7.9 guest only or all rhel7 guest ?

@yanan-fu 

My fault, it is in all rhel7 guests, sorry for that I will modify it, thanks!

File: qemu/tests/hotplug_mem_repeat.py [hunk 1]
```diff
@@ -45,11 +45,18 @@ def start_test(self):
...
+        status, output = session.cmd_status_output(cmd)
+
+        self.test.log.debug("CMD output: %s" % str(output))
+
         if self.params["test_type"] == "scalability_test":
             error_context.context("Repeat hotplug memory for %s times"
                                   % times, LOG_JOB.info)
             self.repeat_hotplug(vm, target_mems)
-            if self.params.get('os_type') == 'linux':
+            if self.params.get('os_type') == 'linux' and status != 0:
```
Comment (huangyum) [inline qemu/tests/hotplug_mem_repeat.py:59]: Instead of 'status', prefer a clearer and meaningful variant name.

Comment (mcasquer) [reply qemu/tests/hotplug_mem_repeat.py:59]: What about something like: **rhel_version_cmd_status**? according also to previous variable **rhel_version**

File: qemu/tests/hotplug_mem_repeat.py [hunk 2]
```diff
@@ -45,11 +45,18 @@ def start_test(self):
             arg = "movable_node"
             utils_test.update_boot_option(vm, args_added=arg)
         original_mem = self.get_guest_total_mem(vm)
+
+        rhel_version = "\"Red Hat Enterprise Linux Server release 7\""
+        cmd = "grep %s /etc/redhat-release" % rhel_version
+        status, output = session.cmd_status_output(cmd)
```
Comment (huangyum) [inline qemu/tests/hotplug_mem_repeat.py:51]: Suggest to move these lines into `if` block in L44.

File: qemu/tests/hotplug_mem_repeat.py (hunk 1 above)
Comment (huangyum) [reply qemu/tests/hotplug_mem_repeat.py:59]: How about keep `status` in L51 and introduce a new one, like `is_rhel_7` ?

Comment (mcasquer) [reply qemu/tests/hotplug_mem_repeat.py]: Updated

File: qemu/tests/hotplug_mem_repeat.py (hunk 2 above)
Comment (mcasquer) [reply qemu/tests/hotplug_mem_repeat.py]: Updated

**Comment by yanan-fu** [comment]
Suggest:
1. cfg:
```
+    Linux:
+        mem_unplug_test = yes
+        RHEL.7:
+            mem_unplug_test = no
```

2. py
```
-            if self.params.get('os_type') == 'linux':
+            if self.params.get_boolean(mem_unplug_test, False):
                 error_context.context("Repeat unplug memory for %s times"
                                       % times, LOG_JOB.info)
                 self.repeat_unplug(vm, target_mems)
```

**Comment by huangyum** [comment]
> Suggest:
> 
>     1. cfg:
> 
> 
> ```
> +    Linux:
> +        mem_unplug_test = yes
> +        RHEL.7:
> +            mem_unplug_test = no
> ```
> 
>     2. py
> 
> 
> ```
> -            if self.params.get('os_type') == 'linux':
> +            if self.params.get_boolean(mem_unplug_test, False):
>                  error_context.context("Repeat unplug memory for %s times"
>                                        % times, LOG_JOB.info)
>                  self.repeat_unplug(vm, target_mems)
> ```

If a new param is introduced to indicate if run unplug test, I guess we don't need to get os_type but set the param to false for windows guest?

**Comment by yanan-fu** [comment]
> 
> If a new param is introduced to indicate if run unplug test, I guess we don't need to get os_type but set the param to false for windows guest?

Yes, agree with that, all actions for unplug only can use the new parameter instead of os_type.  Not only the example pasted above.

**Comment by mcasquer** [comment]
@yanan-fu @huangyum 

Ok so if I understood well:
1.-  Add the new `mem_unplug_test` param in the cfg
2.- Set it as false for rhel7 and Windows guests
3.- Replace the os_type if condition by the one of this new param for unplug actions

That's enough?

**Comment by yanan-fu** [comment]
> @yanan-fu @huangyum
> 
> Ok so if I understood well: 1.- Add the new `mem_unplug_test` param in the cfg 2.- Set it as false for rhel7 and Windows guests 3.- Replace the os_type if condition by the one of this new param for unplug actions
> 
> That's enough?

Yes, it is.  In short, include 4 parts as i can see:
1.   movable_node
2.   repeat_unplug
3.   unplug_memory
4.  compare current_mem and original_mem

**Comment by mcasquer** [comment]
@yanan-fu could you pls review again the PR? Thanks!

File: qemu/tests/cfg/hotplug_mem_repeat.cfg [hunk 3]
```diff
@@ -11,6 +11,12 @@
     mem_fixed = 4096
     size_mem = 128M
     maxmem_mem = 40G
+    Linux:
+        mem_unplug_test = yes
+        RHEL.7:
+            mem_unplug_test = no
+    Windows:
+        mem_unplug_test = no
```
Comment (yanan-fu) [inline qemu/tests/cfg/hotplug_mem_repeat.cfg:19]: How about remove L18~19 as it is the default value ?

Comment (mcasquer) [reply qemu/tests/cfg/hotplug_mem_repeat.cfg]: Done!

**Review by yanan-fu — APPROVED** [review]
Ack

**Pull request merged**